### PR TITLE
Refactor handoff handler

### PR DESCRIPTION
**Pull request by yah01**
related to #17804 #17627 
Change all the handoff task to scheduler style, fix the issue some compacted segments will miss notification

Signed-off-by: yah01 <yang.cen@zilliz.com>
Co-authored-by: xiaofan-luan <xiaofan.luan@zilliz.com>

### CODE REVIEW

**Comment by xige-16** [comment]
If semgent a is compacted during handoff (a, b -> c), the growing semgent a on querynode is not cleaned up in time, then querynode will receive segmentchannelInfo(offline segment={a, b}) when handoff segment c , at this time, the growing semgent a should be cleaned up according to the offline segment.

File: internal/querycoord/meta.go [hunk 1]
```diff
@@ -662,25 +643,44 @@ func (m *MetaReplica) saveGlobalSealedSegInfos(saves col2SegmentInfos) (col2Seal
 	for _, changeInfos := range col2SegmentChangeInfos {
```
Comment (xiaofan-luan) [inline internal/querycoord/meta.go:643]: put before segment meta changed so it could be retried

File: internal/querycoord/handoff_handler.go [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+		}
+		handler.triggerHandoff(task)
+	}
+	// handoffTaskTriggered state don't need to be handled in the loop, it will handled by the go routine in triggerHandoff
+
+	return nil
+}
+
+func (handler *HandoffHandler) clean(segmentID int64) {
+	task := handler.tasks[segmentID]
```
Comment (xiaofan-luan) [inline internal/querycoord/handoff_handler.go:289]: should we clean ChangeInfoMetaPrefix here?

Comment (yah01) [reply internal/querycoord/handoff_handler.go:289]: > should we clean ChangeInfoMetaPrefix here?

QueryNode cleans it, a new shard leader that NodeDown balanced to, can't capture it if we clean it here

**Comment by yah01** [comment]
rerun ut

**Pull request merged**